### PR TITLE
Boost tensor fields

### PR DESCRIPTION
**Pull request by pandu-k**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
No way to boost/anti-boost fields

* **What is the new behavior (if this is a feature change)?**
Now a way to boost fields relative to eachother, during search

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
In progress

* **Related Python client changes** (link commit/PR here)
TODO

* **Related documentation changes** (link commit/PR here)
Boost: https://gh-previews.marqo.pages.dev/0.0.13/API-Reference/search/#boost

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: tests/tensor_search/test_boost_field_scores.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+from marqo.errors import IndexNotFoundError, InvalidArgError
+from marqo.tensor_search import tensor_search
+
+from tests.marqo_test import MarqoTestCase
+
+
+class TestBoostFieldScores(MarqoTestCase):
```
Comment (pandu-k) [inline tests/tensor_search/test_boost_field_scores.py:7]: Can we test with more documents (10 or so) with varying fields, with varying other params (like pagination)?

**Review by jn2clark — APPROVED** [review]

**Comment by wanliAlex** [comment]
image
pagination, use [1,1] as boost dic for both fields.

**Comment by wanliAlex** [comment]
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 1
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 2
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 3
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 4
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 5
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 6
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 7
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 8
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 9
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 10
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 11
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 12
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 13
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 14
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 15
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 16
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 17
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 18
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
Description
before boost:0.7821903
after boost:1.7821903
Title
before boost:0.9306991
after boost:1.9306991
DELETNE to be bootsed
Number of doc 19

different number of docs

**Comment by wanliAlex** [comment]
Title
before boost:0.9306991
after boost:1.9306991
void_field_3
before boost:0.49232623
after boost:1.49232623
Description
before boost:0.7821903
after boost:1.7821903
void_field_8
before boost:0.48269674
after boost:1.48269674
void_field_6
before boost:0.4840921
after boost:1.4840921
void_field_1
before boost:0.48407263
after boost:1.48407263
void_field_5
before boost:0.48731607
after boost:1.4873160699999999
void_field_4
before boost:0.4855699
after boost:1.4855699
void_field_7
before boost:0.4828252
after boost:1.4828252

different number of fields, use boost [1,1]

**Comment by wanliAlex** [comment]
void_field_1
before boost:0.48407263
after boost:-1.48407263
void_field_5
before boost:0.48731607
after boost:-1.4873160699999999
void_field_7
before boost:0.4828252
after boost:-1.4828252
void_field_6
before boost:0.4840921
after boost:-1.4840921
void_field_8
before boost:0.48269674
after boost:-1.48269674
void_field_2
before boost:0.48322064
after boost:-1.4832206399999999
Title
before boost:0.9306991
after boost:0.0
Description
before boost:0.7821903
after boost:1.7821903
void_field_3
before boost:0.49232623
after boost:-1.49232623
void_field_4
before boost:0.4855699
after boost:-1.4855699
void_field_1
before boost:0.48407263
after boost:-1.48407263
void_field_5
before boost:0.48731607
after boost:-1.4873160699999999
void_field_7
before boost:0.4828252
after boost:-1.4828252
void_field_6
before boost:0.4840921
after boost:-1.4840921
void_field_8
before boost:0.48269674
after boost:-1.48269674
void_field_2
before boost:0.48322064
after boost:-1.4832206399999999
Title
before boost:0.9306991
after boost:0.0
Description
before boost:0.7821903
after boost:1.7821903

title [0,0], description[1,1], void_field[-1,-1]

**Comment by wanliAlex** [comment]
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be bootsed
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
Description
before boost:0.7821903
after boost:4.5643806
Title
before boost:0.9306991
after boost:-4.7920973
DELETNE to be boosted

title: [-3,-2], description [2,3]

**Pull request merged**